src/state/MuteStates.ts

```diff
@@ -135,21 +159,45 @@ class MuteState<Label, Selected> {
     private readonly device: MediaDevice<Label, Selected>,
     private readonly joined$: Observable<boolean>,
     private readonly enabledByConfig: boolean,
+    /**
+     * An optional observable which, when it emits `true`, will force the mute.
+     * Used for video to stop camera when earpiece mode is on.
+     * @private
+     */
+    private readonly forceMute$: Observable<boolean>,
```

Comment:
Maybe this is also what we should use if we run have an error for any of the devices. Add the force mute flag and then render the button as in disabled state.

Reply:
Yes we have to check for these errors and when they occur. But maybe now it will just make the `available` map empty, that is already muting. What error are you thinking about?